terraspace/app/stacks/peer/main.tf

```diff
@@ -0,0 +1,172 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.38"
+    }
+
+    http = {
+      source  = "hashicorp/http"
+      version = "~> 2.1"
+    }
+  }
+  required_version = ">= 1.0.0"
+}
+
+data "aws_eks_cluster" "eks" {
+  name = module.eks.cluster_id
+}
+
+data "aws_availability_zones" "available" {
+}
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 3.0"
+
+  name                 = var.vpc.name
+  cidr                 = "10.0.0.0/16"
+  azs                  = data.aws_availability_zones.available.names
+  private_subnets      = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24", "10.0.4.0/24"] # Worker Nodes
+  public_subnets       = ["10.0.5.0/24", "10.0.6.0/24", "10.0.7.0/24"]                # LoadBalancer and NAT
+  enable_nat_gateway   = true
+  single_nat_gateway   = true
+  enable_dns_hostnames = true
+
+  public_subnet_tags = {
+    "kubernetes.io/cluster/${var.cluster_name}" = "shared"
+    "kubernetes.io/role/elb"                    = "1"
+  }
+
+  private_subnet_tags = {
+    "kubernetes.io/cluster/${var.cluster_name}" = "shared"
+    "kubernetes.io/role/internal-elb"           = "1"
+  }
+}
+
+module "gateway_endpoint_to_dynamodb" {
+  source         = "../../modules/gateway-endpoint-to-dynamodb"
+  vpc_id         = module.vpc.vpc_id
+  region         = var.region
+  route_table_id = module.vpc.private_route_table_ids[0]
+}
+
+module "gateway_endpoint_to_s3" {
+  source         = "../../modules/gateway-endpoint-to-s3"
+  vpc_id         = module.vpc.vpc_id
+  region         = var.region
+  route_table_id = module.vpc.private_route_table_ids[0]
+}
+
+data "http" "myip" {
+  url = "http://ipv4.icanhazip.com"
+}
+
+module "eks" {
+  source                             = "terraform-aws-modules/eks/aws"
+  version                            = "~> 18.2.0"
+  cluster_name                       = var.cluster_name
+  cluster_version                    = var.cluster_version
+  cluster_endpoint_private_access    = true
+  cluster_endpoint_public_access     = true
+  vpc_id                             = module.vpc.vpc_id
+  subnet_ids                         = [module.vpc.private_subnets[0], module.vpc.private_subnets[1]]
+  enable_irsa                        = true # To be able to access AWS services from PODs  
+  cluster_security_group_description = "EKS cluster security group - Control Plane"
+
+  cluster_endpoint_public_access_cidrs = [ # TODO: Access through AWS transit gateway
+    "${chomp(data.http.myip.body)}/32",    # GitHub Actions Self Runner Static IP 
+    "177.33.141.81/32",
+    "185.152.47.29/32",
+    "168.227.34.17/32",
+  ]
+
+  eks_managed_node_groups = { # Needed for CoreDNS (https://docs.aws.amazon.com/eks/latest/userguide/fargate-getting-started.html)
+    test-ipfs-peer-subsys = {
+      name         = var.cluster_name
+      desired_size = 2
+      min_size     = 2
+      max_size     = 20
+
+      instance_types = ["c6i.2xlarge"]
```

Comment:
this should be configurable, at least at the env level.

Reply:
Done